tsdb/engine/tsm1/file_store.go

```diff
@@ -284,22 +278,60 @@ func (f *FileStore) WithLogger(log *zap.Logger) {
 	}
 }
 
-// FileStoreStatistics keeps statistics about the file store.
-type FileStoreStatistics struct {
-	DiskBytes int64
-	FileCount int64
+var globalFileStoreMetrics = newAllFileStoreMetrics()
+
+const namespace = "storage"
+const filesSubsystem = "tsm_files"
+
+type allFileStoreMetrics struct {
+	files *prometheus.GaugeVec
+	size  *prometheus.GaugeVec
+}
+
+type fileStoreMetrics struct {
+	files      prometheus.Gauge
+	size       prometheus.Gauge
+	sizeAtomic int64
```

Comment:
How does this value get initialized when for ex. the server first starts up? Does that process go through `(f *FileStore) replace()` also?

Reply:
In `replace`, we call SetFiles and SetSize to reset the values.

In `Open`, we call SetFiles and AddSize as we open the store.

This is a fairly mechanical translation from what was here before, with atomic stores and adds.